neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/impl/BoltNeo4jConnectionImpl.java

```diff
@@ -353,15 +359,16 @@ private void initSession() {
 
     private void initTransaction()  {
 	    try {
-            if (this.transaction == null) {
-                this.transaction = this.session.beginTransaction();
-			}
-            else if (this.getAutoCommit()) {
+			if (this.getAutoCommit()) {
```

Comment:
do we call this after a statement or before the next?

if before the next, does this mean that a commit can be delayed by a long time if no 2nd statement follows?

Reply:
@jexp what do you mean?

Reply:
the method sounds as if it is called at the beginning of a statement.
and then closes teh *previous* session.

But that would also mean that if an statement executed, and then the next statement is only executed minutes or hours later,the session would stay open that long, or?